src/main/java/com/github/seratch/jslack/api/model/block/ActionsBlock.java

```diff
@@ -0,0 +1,23 @@
+package com.github.seratch.jslack.api.model.block;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.github.seratch.jslack.api.model.block.element.BlockElement;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+/**
+ * https://api.slack.com/reference/messaging/blocks#actions
+ */
+@Data
+@EqualsAndHashCode(callSuper = true)
+@Builder
```

Comment:
Regarding any classes under the `api.model` package, I intentionally avoid using `@Builder` annotation because just having it results in skipping generating default constructors for the class. 

Most JSON parsing libraries (we currently use gson internally) expect non-arg constructors. Therefore, if we simply have `@Builder`, a problem similar to the following JPA related one may happen. https://stackoverflow.com/questions/40516058/using-lomboks-data-and-builder-on-entity

I would like to explicitly have NoArgsConstructor/AllArgsConstructor as a workaround.

Reply:
Also, I prefer @Builder(toBuilder = true). https://projectlombok.org/features/Builder

Reply:
I've added the `@NoArgsConstructor @AllArgsConstructor` to the classes. 

I have not found a single use of `toBuilder=true` in the code base _yet_. I would like to add this in a new issue (e.g. "Add Lombok's toBuilder for easier templating of model objects").

I looked at the `api.model` package and found many `@Builder` annotations. I don't think I can find a particular pattern in which classes should/shouldn't have `@Builder` annotations.

- `@Data @Builder public class Dialog`
- `@Data @Builder public class DialogOption`
- `@Data public class DialogResponseMetadata`
- `@Data @Builder public class DialogSelectElement`
- `@Data @Builder public class Action`
- `@Data @Builder public class Attachment`
- `@Data public class BillableInfo`

I'm not criticising, but trying to understand when to apply `@Builder` and when not. Is the distinction that for objects that are returned from the Slack API you don't apply `@Builder`, but for objects that are only sent so Slack in a payload you do?

Reply:
Okay, you're right. The above comment is my intention for the future. never mind.